server/server.go

```diff
@@ -1372,12 +1360,34 @@ func (s *Server) Start() {
 		s.checkResolvePreloads()
 	}
 
-	// In operator mode, when the account resolver depends on an external system and
-	// the system account is the bootstrapping account, start fetching it
-	if ar := s.accResolver; len(opts.TrustedOperators) == 1 && ar != nil &&
-		opts.SystemAccount != _EMPTY_ && opts.SystemAccount != DEFAULT_SYSTEM_ACCOUNT {
-		if _, ok := ar.(*MemAccResolver); !ok {
-			if v, ok := s.accounts.Load(s.opts.SystemAccount); ok && v.(*Account).claimJWT == "" {
+	// Log the pid to a file
+	if opts.PidFile != _EMPTY_ {
+		if err := s.logPid(); err != nil {
+			PrintAndDie(fmt.Sprintf("Could not write pidfile: %v\n", err))
+		}
+	}
+
+	// Setup system account which will start the eventing stack.
+	if sa := opts.SystemAccount; sa != _EMPTY_ {
+		if err := s.SetSystemAccount(sa); err != nil {
+			s.Fatalf("Can't set system account: %v", err)
+			return
+		}
+	} else if !opts.NoSystemAccount {
+		// We will create a default system account here.
+		s.SetDefaultSystemAccount()
+	}
+
+	// start up resolver machinery
+	if ar := s.AccountResolver(); ar != nil {
+		if err := ar.Start(s); err != nil {
+			PrintAndDie(fmt.Sprintf("Could not start resolver: %v\n", err))
+		}
+		// In operator mode, when the account resolver depends on an external system and
+		// the system account is the bootstrapping account, start fetching it
+		if len(opts.TrustedOperators) == 1 && opts.SystemAccount != _EMPTY_ && opts.SystemAccount != DEFAULT_SYSTEM_ACCOUNT {
+			if _, ok := ar.(*MemAccResolver); !ok {
+			} else if v, ok := s.accounts.Load(s.opts.SystemAccount); ok && v.(*Account).claimJWT == "" {
```

Comment:
Same for line 1401:
```
if _, err := ar.Fetch(s.opts.SystemAccount); err != nil {
} else if _, err := s.fetchAccount(s.opts.SystemAccount); err == nil {
    s.Noticef("System account fetched and updated")
    return
}
```
I personally would read better:
```
if _, err := ar.Fetch(s.opts.SystemAccount); err != nil {
    // Let's try again at the next tick.
    continue
}
if _, err := s.fetchAccount(s.opts.SystemAccount); err == nil {
    s.Noticef("System account fetched and updated")
    return
}
```

Reply:
fixed